File: sprint5/go/lesson7code1.go
type Node struct {
    key    int
    height int
    left   *Node
    right  *Node
}

func NewNode(k int, h int, l *Node, r *Node) *Node {
    return &Node{key: k, height: h, left: l, right: r}
}

func NewNode(k int) *Node {
    return NewNode(k, 1, nil, nil)
}

File: sprint5/kotlin/lesson7code1.kt
class Node {
    var key: Int
    var height: Int = 1
    var left: Node? = null
    var right: Node? = null

    constructor(key: Int, height: Int = 1, left: Node? = null, right: Node? = null) {
        this.key = key
        this.height = height
        this.left = left
        this.right = right
    }
}
